lightning/src/ln/chan_utils.rs

```diff
@@ -265,7 +265,7 @@ pub fn derive_private_key<T: secp256k1::Signing>(secp_ctx: &Secp256k1<T>, per_co
 	let res = Sha256::from_engine(sha).into_inner();
 
 	let mut key = base_secret.clone();
-	key.add_assign(&res)?;
+	key = key.add_tweak(&Scalar::from_be_bytes(res).unwrap())?;
```

Comment:
You could get rid of `mut` and an extra line if you write `let key = base_secret.add_tweak(&Scalar::from_be_bytes(res).unwrap())?;`

Reply:
will do

Reply:
it ended up that many of the intermediate variables were superfluous anyway